hyperspy/io_plugins/blockfile.py

```diff
@@ -270,9 +322,13 @@ def file_writer(filename, signal, **kwds):
         # Write virtual bright field
         vbf = signal.mean(
             signal.axes_manager.signal_axes[
-                :2]).data.astype(
-            endianess +
-            'u1')
+                :2])
+        if signal._lazy:
+            vbf.compute()
```

Comment:
I implemented something similar (if `None` -> array of zeros, if `'navigator'` -> navigator for lazy dataset, mean for non-lazy, if a signal is passed, signal is used.) but it would be cleaner and benefit from a resolution of  #2639